src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java

```diff
@@ -75,24 +76,33 @@ public void importEntries(List<BibEntry> entriesToImport) {
         // Check if we are supposed to warn about duplicates.
         // If so, then see if there are duplicates, and warn if yes.
         if (preferences.shouldWarnAboutDuplicatesForImport()) {
-            boolean containsDuplicate = entriesToImport.stream()
-                                                       .anyMatch(this::hasDuplicate);
-
-            if (containsDuplicate) {
-                boolean continueImport = dialogService.showConfirmationDialogWithOptOutAndWait(Localization.lang("Duplicates found"),
-                        Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
-                        Localization.lang("Continue with import"),
-                        Localization.lang("Cancel import"),
-                        Localization.lang("Disable this confirmation dialog"),
-                        optOut -> preferences.setShouldWarnAboutDuplicatesForImport(!optOut));
-
-                if (!continueImport) {
-                    dialogService.notify(Localization.lang("Import canceled"));
-                    return;
+            BackgroundTask.wrap(() -> entriesToImport.stream()
+                    .anyMatch(this::hasDuplicate)).onSuccess(duplicateFound -> {
+                if (duplicateFound) {
+                    boolean continueImport = dialogService.showConfirmationDialogWithOptOutAndWait(Localization.lang("Duplicates found"),
+                            Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
+                            Localization.lang("Continue with import"),
+                            Localization.lang("Cancel import"),
+                            Localization.lang("Disable this confirmation dialog"),
+                            optOut -> preferences.setShouldWarnAboutDuplicatesForImport(!optOut));
+
+                    if (!continueImport) {
+                        dialogService.notify(Localization.lang("Import canceled"));
+                    } else {
+                        buildImportHandlerThenImportEntries(entriesToImport);
+                        dialogService.notify(Localization.lang("Number of entries successfully imported") + ": " + entriesToImport.size());
```

Comment:
Please move the `notify` to the end of `buildImportHandlerThenImportEntries` so that the message is shown in every case after the import.

Reply:
sure, should be done